src/EditorFeatures/Test/Diagnostics/TestDiagnosticAnalyzerService.cs

```diff
@@ -40,8 +42,9 @@ internal TestDiagnosticAnalyzerService(ImmutableArray<AnalyzerReference> hostAna
             HostAnalyzerManager hostAnalyzerManager,
             AbstractHostDiagnosticUpdateSource hostDiagnosticUpdateSource,
             Action<Exception, DiagnosticAnalyzer, Diagnostic> onAnalyzerException,
-            IDiagnosticUpdateSourceRegistrationService registrationService = null)
-            : base(hostAnalyzerManager, hostDiagnosticUpdateSource, registrationService ?? new MockDiagnosticUpdateSourceRegistrationService())
+            IDiagnosticUpdateSourceRegistrationService registrationService = null,
+            IAsynchronousOperationListener listener = null)
```

Comment:
Regardless, because of this, we still have incredibly difficult issues to track down.  The feature is *conditionally* optional.  Optional parameters should only be used when the value is truly optional and will cause no problems if the caller forgets to pass them.